File: lua/cycle_a.lua
-- C1.a — Lua cycle via package.loaded caching.
-- When cycle_a requires cycle_b and cycle_b tries to require cycle_a,
-- Lua returns the partially-initialised module table of cycle_a.
-- The LSP has to handle that partial-load state gracefully.

local M = {}

function M.describe(a)    return "Alpha(" .. a.name .. ")" end
function M.spawn_bravo(a)
    local cycle_b = require("cycle_b")        -- lazy require inside fn
    return cycle_b.new(a.name .. "/b")
end
function M.kick_off()
    local a = { name = "root" }
    local b = M.spawn_bravo(a)
    local cycle_b = require("cycle_b")
    return cycle_b.bounce_to_alpha(b)
end

return M

File: monorepo_java/apps/web/src/main/java/mono/web/CycleA.java
package mono.web;

// C1.a — Java cycle. Same package, no .java forward-decl needed.
public class CycleA {
    private final String name;
    private CycleB child;                        // C1.a.type_ref → CycleB

    public CycleA(String name) { this.name = name; }

    public CycleB spawnBravo() { return new CycleB(name + "/b"); }
    public String describe()   { return "CycleA(" + name + ")"; }

    public static String kickOff() {
        CycleA a = new CycleA("root");
        CycleB b = a.spawnBravo();
        return b.bounceToAlpha();
    }
}
